usecases/cem/cdt/public.go

```diff
@@ -0,0 +1,164 @@
+package cdt
+
+import (
+	"github.com/enbility/eebus-go/api"
+	"github.com/enbility/eebus-go/features/client"
+	usecasesapi "github.com/enbility/eebus-go/usecases/api"
+	"github.com/enbility/ship-go/logging"
+	spineapi "github.com/enbility/spine-go/api"
+	"github.com/enbility/spine-go/model"
+)
+
+// Setpoints retrieves the setpoints for various HVAC operation modes from a remote entity.
+//
+// Possible errors:
+//   - ErrDataNotAvailable: If the mapping of operation modes to setpoints or the setpoints themselves are not available.
+//   - Other errors: Any other errors encountered during the process.
+func (e *CDT) Setpoints(entity spineapi.EntityRemoteInterface) ([]usecasesapi.Setpoint, error) {
+	setpoints := make([]usecasesapi.Setpoint, 0)
+
+	sp, err := client.NewSetpoint(e.LocalEntity, entity)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, setpoint := range sp.GetSetpoints() {
+		var value float64 = 0
+		var minValue float64 = 0
+		var maxValue float64 = 0
+		var timePeriod model.TimePeriodType = model.TimePeriodType{}
+
+		if setpoint.SetpointId == nil {
+			logging.Log().Error("Setpoint ID is nil")
+			continue
+		}
+
+		if setpoint.Value != nil {
+			value = setpoint.Value.GetValue()
+		}
+
+		if setpoint.ValueMax != nil {
+			maxValue = setpoint.ValueMax.GetValue()
+		}
+
+		if setpoint.ValueMin != nil {
+			minValue = setpoint.ValueMin.GetValue()
+		}
+
+		if setpoint.TimePeriod != nil {
+			timePeriod = *setpoint.TimePeriod
+		}
+
+		isActive := (setpoint.IsSetpointActive == nil || *setpoint.IsSetpointActive)
```

Comment:
According to the spec. the following rules apply:

- `isSetpointChangeable` - If set to "true" the server SHALL accept changes of the setpoint by appropriate clients (e.g. the bound client). If set to "false" the server SHALL decline change requests. If absent, the server SHOULD accept changes.

- `isSetpointActive` - If set to "false", the setpoint is inactive. If set to "true" or omitted, the setpoint is active.

Ref: _Resource Specification_: _4.3.23.4_ `setpointListData`.

Reply:
Is this a question or a note to yourself? :)

Reply:
This is a note for you TBH 🙂 

To clarify why I used the || operator:

In some parts of EEBus, for boolean fields, an absent (`nil`) value is considered false, so you typically check it like this: `isActive := (setpoint.IsSetpointActive != nil && *setpoint.IsSetpointActive)`

However, in this case, the specification requires the opposite: if the value is absent, it should default to `true`.

Reply:
Hehe 😃 

Yeah, consistency usually wins, and hurts when it is not present. Would you mind adding this note as a comment into the code? It is nearly impossible to keep this in mind and always check the specs or find this note.

Reply:
Done. 

Added this note as a comment into the code:
```Go
// As per [Resource Specification] 4.3.23.4 setpointListData:
// - isSetpointActive: If false, the setpoint is inactive; if true or omitted, it is active.
// - isSetpointChangeable: If true, the server accepts changes; if false, it declines changes. If absent, changes are accepted.
isActive := (setpoint.IsSetpointActive == nil || *setpoint.IsSetpointActive)
isChangeable := (setpoint.IsSetpointChangeable == nil || *setpoint.IsSetpointChangeable)
```